qiita_plugins/target_gene_type/target_gene_type/validate.py

```diff
@@ -0,0 +1,469 @@
+            for old in f:
+                f.move(old, id_map[old])
+
+        # When we fix, we always generate the FASTQ and FASTA file
+        # By setting them to None, below will be generated
+        demux_fp = new_demux_fp
+        fastq_fp = None
+        fasta_fp = None
+
+    # If we didn't fix anything, we only generate the files if they don't
+    # already exists
+    name = splitext(basename(demux_fp))[0]
+    if not fastq_fp:
+        fastq_fp = join(out_dir, "%s.fastq" % name)
+        with open(fastq_fp, 'w') as fq:
+            with File(demux_fp, 'r') as dx:
+                for record in to_ascii(dx):
+                    fq.write(record)
+
+    if not fasta_fp:
+        fasta_fp = join(out_dir, "%s.fasta" % name)
+        with open(fasta_fp, 'w') as f:
+            for r in load(fastq_fp):
+                f.write(format_fasta_record(r['SequenceID'], r['Sequence'],
+                                            r['Qual']))
+
+    filepaths = [[[fastq_fp], 'preprocessed_fastq'],
+                 [[fasta_fp], 'preprocessed_fasta'],
+                 [[demux_fp], 'preprocessed_demux']]
+    if log_fp:
+        filepaths.append([[log_fp], 'log'])
+    return format_payload(
+        success=True, artifacts_info=[[None, 'Demultiplexed', filepaths]])
+
+
+def _validate_demultiplexed(qclient, job_id, prep_info, files, out_dir):
+    """Validate and fix a new 'Demultiplexed' artifact
+
+    Parameters
+    ----------
+    qclient : qiita_client.QiitaClient
+        The Qiita server client
+    job_id : str
+        The job id
+    prep_info : dict of {str: dict of {str: str}}
+        The prep information keyed by sample id
+    files : dict of {str: list of str}
+        The files to add to the new artifact, keyed by filepath type
+    out_dir : str
+        The output directory
+
+    Returns
+    -------
+    dict
+        The results of the job
+    """
+    qclient.update_job_step(job_id, "Step 2: Validating 'Demultiplexed' files")
+
+    supported_fp_types = {'preprocessed_fasta', 'preprocessed_fastq',
+                          'preprocessed_demux', 'log'}
+    unsupported_fp_types = set(files) - supported_fp_types
+    if unsupported_fp_types:
+        return format_payload(
+            success=False,
+            error_msg="Filepath type(s) %s not supported by artifact type "
+                      "Demultiplexed. Supported filepath types: %s"
+                      % (', '.join(unsupported_fp_types),
+                         ', '.join(sorted(supported_fp_types)))
+        )
+
+    # At most one file of each type can be provided
+    offending = set(fp_t for fp_t, fps in files.items() if len(fps) > 1)
+    if offending:
+        errors = ["%s (%d): %s"
+                  % (fp_t, len(files[fp_t]), ', '.join(files[fp_t]))
+                  for fp_t in sorted(offending)]
+        return format_payload(
+            success=False,
+            error_msg="Only one file of each filepath type is supported. %s"
```

Comment:
`Only one file of each filepath type is supported` -> `Only one filepath of each file type is supported, the following files .... `. Also perhaps worth adding a new line at the end of this sentence.